File: backend-java/src/main/java/com/campus/literature/entity/LlmConfig.java
package com.campus.literature.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * LLM API 配置实体
 */
@Data
@TableName("llm_config")
public class LlmConfig {

    @TableId(type = IdType.AUTO)
    private Long id;
    private String name;
    private String provider;
    private String baseUrl;
    private String model;
    private String apiKey;
    private Integer enabled;
    private Integer active;
    private Integer timeoutSeconds;
    private String remark;
    private LocalDateTime createTime;
    private LocalDateTime updateTime;
}

File: frontend/src/api/llm-config.ts
import { request } from "./request";

export interface LlmConfig {
  id: number;
  name: string;
  provider: string;
  baseUrl: string;
  model: string;
  apiKeyMasked: string;
  enabled: number;
  active: number;
  timeoutSeconds: number;
  remark: string;
  createTime: string;
  updateTime: string;
}

export interface LlmConfigForm {
  name: string;
  provider: string;
  baseUrl: string;
  model: string;
  apiKey: string;
  enabled: number;
  active: number;
  timeoutSeconds: number;
  remark: string;
}

export const llmConfigApi = {
  list: async (): Promise<LlmConfig[]> => {
    return request<LlmConfig[]>({ method: "GET", url: "/admin/llm-configs" });
  },

  getActive: async (): Promise<LlmConfig | null> => {
    return request<LlmConfig | null>({ method: "GET", url: "/admin/llm-configs/active" });
  },

  create: async (data: LlmConfigForm): Promise<LlmConfig> => {
    return request<LlmConfig>({ method: "POST", url: "/admin/llm-configs", data });
  },

  update: async (id: number, data: LlmConfigForm): Promise<LlmConfig> => {
    return request<LlmConfig>({ method: "PUT", url: `/admin/llm-configs/${id}`, data });
  },

  remove: async (id: number): Promise<void> => {
    return request<void>({ method: "DELETE", url: `/admin/llm-configs/${id}` });
  },

  activate: async (id: number): Promise<void> => {
    return request<void>({ method: "POST", url: `/admin/llm-configs/${id}/activate` });
  },

  test: async (id: number): Promise<string> => {
    const res = await request<any>({ method: "POST", url: `/admin/llm-configs/${id}/test` });
    return res?.data || "";
  },
};
